karabo_bridge/client.py

```diff
@@ -116,3 +116,15 @@ def _deserialize(self, msg):
             else:
                 raise RuntimeError('unknown message content:', md['content'])
         return dat
+
+    def __enter__(self):
+        return self
+    
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
```

Comment:
Should this close the socket or something? There's no point having a context manager if neither enter nor exit does anything.

Reply:
True, maybe it is useless.
I also thought that the socket would be closed automatically when the context is excited, isn't it the case?

Reply:
Not automatically from our point of view! All the `with` statement does is call these `__enter__` and `__exit__` methods. If we want it to have any effect, we have to write it in these methods.

It probably makes sense for it to close the socket on exit; I just checked, and `with ctx.socket() as sock:` will close on exiting the context as well. In most use cases I see, you'll exit the context when you're about to exit the program, so the socket should be cleaned up anyway. But it doesn't hurt to have an explicit way to ensure it's cleaned up.